File: OOPS/sessional 2/passingparameters.cpp
//         Book(const string &title, int id, const string &author) : 
//         LibraryItem(title, id) {
//         this->author=author; }

#include <iostream>
#include <string>
using namespace std;

// Base class
class LibraryItem {
protected:
    string title;
    int id;

public:
    // Base class constructor
    LibraryItem(const string &title, int id) {
  this->id=id;
  this->title = title;
  cout << "LibraryItem Constructor: " << title << " (ID: " << id << ")" << endl;
    } 
    
    void display() {
        cout << this->id <<endl <<this->title <<endl;
    }

};

// Derived class for Book
class Book : public LibraryItem {
private:
    string author;

public:
    // Derived class constructor
    Book(const string &title, int id, const string &author) : LibraryItem(title, id) {
        this->author=author;
        cout << "Book Constructor: " << title << " by " << author << endl;
    }
    void display() {
    LibraryItem :: display() ; // calling base fn from derived
    cout << this->author <<endl <<endl;
    }


};

// Derived class for Magazine
class Magazine : public LibraryItem {
private:
    int issue;

public:
    // Derived class constructor
    Magazine(const string &title, int id, int issue) : LibraryItem(title, id), issue(issue) {
        cout << "Magazine Constructor: " << title << " (Issue: " << issue << ")" << endl;
    }
};

// Derived class for Newspaper
class Newspaper : public LibraryItem {
private:
    string date;

public:
    // Derived class constructor
    Newspaper(const string &title, int id, const string &date) : LibraryItem(title, id), date(date) {
        cout << "Newspaper Constructor: " << title << " (Date: " << date << ")" << endl;
    }
};

int main() {
    Book book("The Great Gatsby", 1, "F. Scott Fitzgerald");
    book.display();
    Magazine magazine("National Geographic", 2, 2023);
    Newspaper newspaper("The New York Times", 3, "2024-10-17");

    return 0;
}

File: OOPS/sessional 2/passingparameters.java
class LibraryItem {
    protected String title;
    protected int id;

    // Base class constructor
    public LibraryItem(String title, int id) {
        this.title = title;
        this.id = id;
        System.out.println("LibraryItem Constructor: " + title + " (ID: " + id + ")");
    }
}

// Derived class for Book
class Book extends LibraryItem {
    private String author;

    // Derived class constructor
    public Book(String title, int id, String author) {
        super(title, id); // Call base class constructor
        this.author = author;
        System.out.println("Book Constructor: " + title + " by " + author);
    }
}

// Derived class for Magazine
class Magazine extends LibraryItem {
    private int issue;

    // Derived class constructor
    public Magazine(String title, int id, int issue) {
        super(title, id); // Call base class constructor
        this.issue = issue;
        System.out.println("Magazine Constructor: " + title + " (Issue: " + issue + ")");
    }
}

// Derived class for Newspaper
class Newspaper extends LibraryItem {
    private String date;

    // Derived class constructor
    public Newspaper(String title, int id, String date) {
        super(title, id); // Call base class constructor
        this.date = date;
        System.out.println("Newspaper Constructor: " + title + " (Date: " + date + ")");
    }
}

public class passingparameters {
    public static void main(String[] args) {
        Book book = new Book("The Great Gatsby", 1, "F. Scott Fitzgerald");
        Magazine magazine = new Magazine("National Geographic", 2, 2023);
        Newspaper newspaper = new Newspaper("The New York Times", 3, "2024-10-17");
    }
}




// class base {

//     base() {
//         super (a,b);
//         this->c=c;
//     }
// }

